# Determination - Deterministic rendering environment for white-axe's music
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# ZynAddSubFX's PADsynth normally spawns multiple threads to speed up the
# generation of the samples. However, that makes the generated samples
# nondeterministic, so let's just make sure the samples are generated by
# a single thread instead.

--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -994,7 +994,7 @@ int PADnoteParameters::sampleGenerator(PADnoteParameters::callback cb,
     if(oscilgen->needPrepare())
         oscilgen->prepare();
 
-#ifdef WIN32
+#if true //Concurrency induces nondeterminism! Let's actually not use multithreading on any platform.
     //Temporarily disable multi-threading here as C++11 threads are broken on
     //mingw cross compilation
     thread_cb(1,0);
